content/en/blog/_posts/2020-12-04-gsod-2020-improving-api-reference-experience.md

```diff
@@ -0,0 +1,81 @@
+
+Generally, APIs are used by developers by calling endpoints with a specific HTTP verb, with specific parameters and waiting for a response. The `swagger-ui` interface is built for this usage: the interface displays a list of endpoints and their associated verbs, and for each the parameters and responses formats. 
+
+The Kubernetes API is most of the time used differently: users create manifest files containing resources definitions in YAML format, and use the `kubectl` CLI to *apply* these manifests to the cluster. In this case, the most important information is the description of the structures used as parameters and responses (the Kubernetes Resources).
+
+Because of this specificity, we realized that it would be difficult to adapt the `swagger-ui` interface to satisfy the users of the Kubernetes API.
+
+## Markdown pages
+
+The second stage of the project has been to adapt the work I had done to create the k8sref.io website, to include it in the officiel documentation website.
+
+The main changes have been to:
+- use go-templates to represent the output pages, so non-developers can adapt the generated pages without having to edit the generator code
+- create a shortcode, to easily create links from inside the website to specific pages of the API reference
+- improve the navigation between the sections of the API reference
+- add the code of the generator to the Kubernetes GitHub repository containing the different reference generators
+
+All the discussions and work done can be found on [this Pull Request](https://github.com/kubernetes/website/pull/23294).
+
+The Pull request to add the code of the generator to the Kubernetes GitHub repository is [this Pull Request](https://github.com/kubernetes-sigs/reference-docs/pull/179).
+
+Here are the features of the API Reference included in the official documentation website:
+
+- the resources are categorized, in the categories Workloads, Services, Config & Storage, Authentication, Authorization, Policies, Extend, Cluster. This structure is configurable with a simple [`toc.yaml` file](https://github.com/kubernetes-sigs/reference-docs/blob/master/gen-resourcesdocs/config/v1.20/toc.yaml)
+- each page displays at the first level the associated resources, for example Pod, PodSpec, PodStatus, PodList
+- each resource inlines its definitions (except when definitions are common to several resources, or are too complex to be displayed inline)
+- some widely used definitions are documented in a specific page (ex ObjectMeta)
+- required fields are indicated, and placed first
+- fields of a resource can be categorized and ordered, with the help of a [`fields.yaml` file](https://github.com/kubernetes-sigs/reference-docs/blob/master/gen-resourcesdocs/config/v1.20/fields.yaml)
+- maps fields are indicated (ex pod.spec.nodeSelector is map[string]string, instead of object) using the value of `x-kubernetes-list-type`
```

Comment:
- `map` fields are indicated; for example,  the `.spec.nodeSelector` for a `Pod` is `map[string]string`, instead of `object`) using the value of `x-kubernetes-list-type`